File: orbit-frontend/src/services/ManagerService.js
import api from "./api";

const getAllManagers = async () => {
  const res = await api.get("/managers");
  return res.data;
};

const createManager = async (data) => {
  const res = await api.post("/managers", data);
  return res.data;
};

const deleteManager = async (id) => {
  const res = await api.delete(`/managers/${id}`);
  return res.data;
};

const getManagerByEmail = async (email) => {
  const res = await api.get(`/managers/email/${email}`);
  return res.data;
};

const updateManager = async (id, data) => {
  const res = await api.put(`/managers/${id}`, data);
  return res.data;
};

const managerService = {
  getAllManagers,
  createManager,
  deleteManager,
  updateManager,
  getManagerByEmail,
};

export default managerService;

File: OrbitBackend/src/main/java/com/orbit/service/ManagerService.java
package com.orbit.service;

import java.util.List;
import com.orbit.entity.Manager;

public interface ManagerService {

    Manager saveManager(Manager manager);

    List<Manager> getAllManagers();

    Manager getManagerById(Long id);

    Manager updateManager(Long id, Manager manager);

    void deleteManager(Long id);

    Manager getManagerByEmail(String email);
}